packages/kiota_abstractions/lib/src/api_exception.dart

```diff
@@ -12,13 +12,13 @@ class ApiException implements Exception {
   });
 
   /// The HTTP status code of the response.
-  final int? statusCode;
+  int? statusCode;
```

Comment:
You wrote in the commit message that we shouldn't use `copyWith` because subclasses loose their values when copying.

The `copyWith` pattern is a commonly used pattern in Dart and I'd like to keep it, ideally. The idea is that subclasses should override the `copyWith` with their custom properties.

Does this not work for the use-cases here?

Reply:
I wasn't aware of this pattern, but gave it a shot. I'm not sure if I used it correctly. To me it seems weird to ignore all the arguments of the copywith method. This is how you intended for it to be used? 
```
class CustomError extends ApiException {
  String? someProperty;
  int? someNumber;

  CustomError({
    this.someNumber,
    this.someProperty,
  });

  @override
  ApiException copyWith({
    int? statusCode,
    String? message,
    Map<String, List<String>>? responseHeaders,
    Iterable<Object?>? innerExceptions,
  }) {
    return CustomError(
      someNumber: someNumber, 
      someProperty: someProperty);
  }
}
```

Reply:
Not quite. The `copyWith` method is supposed to take in all the parameters the constructor takes in (but nullable, if not already) and passes all of its properties as fallbacks for the parameters given to `copyWith`:

```dart
class Foo {
    const Foo({required this.val});

    final int val;

    Foo copyWith({
        int? val,
    }) {
        return Foo(val: val ?? this.val);
    }
}
```

With `copyWith` you can change a single property while keeping all other values.

I imagined an override to look like this:

```dart
class Bar extends Foo {
    const Bar({
        required super.val,
        required this.mul,
    });

    final int mul;

    @override
    Bar copyWith({
        int? val,
        int? mul,
    }) {
        return Bar(
            val: val ?? this.val,
            mul: mul ?? this.mul,
        );
    }
}
```